### DISCUSSION TITLE
Credit-Based Shaping - Showcase

### DISCUSSION THREAD

**Discussion by adamgeorge309** [inet-framework/inet#800] [Showcases]
Discuss the "Credit-Based Shaping" showcase here. Ideas and recommendations are welcome.

**Comment by ahmed3009** [comment]
Hi. I am tried to add another traffic class to the showcase by updating relevant parameters such as source, sink apps, another class type and mac queue object etc. However, it is not working. The sub queue for the third class is not containing any packets and the gate and transmission statistics are also not being updated. I can share the code if you want. Any suggestions?

**Reply by PhilippComet** [reply]
I am struggling with the same issue. Did you already find a solution?

**Reply by ahmed3009** [reply]
Yes. Look closely at the pcp values. They need to be set as IEEE standard. https://en.wikipedia.org/wiki/IEEE_P802.1p for reference

**Reply by PhilippComet** [reply]
Thank for the hint! But finally the mapping Table 8-5 "Recommended priority to traffic class mappings" in the IEEE802.1Q-2018 has led me to the solution

**Reply by paixingxing-1** [reply]
I also encountered the same problem, after I added the audio stream, the audio stream has no data and no histogram, how did you solve it

**Reply by ahmed3009** [reply]
@paixingxing-1 As mentioned above, you need to look at the PCP values (Table 8-5 "Recommended priority to traffic class mappings") of the classes that you are setting. If they are incorrect, all audio stream packets will continue to go to the same video stream class.

**Comment by ahmed3009** [comment]
Yes I used the same resource. I just couldn't remember the link to the table you mentioned.

**Comment by Kostas-Tsak** [comment]
Hello could somebody help me on how to plot this image of queue length at the switch? 
Do I need to use TsnLinearNetwork.switch.eth[1].macLayer.queue.queue[0] and TsnLinearNetwork.switch.eth[1].macLayer.queue.queue[1] for this plot?
![image](https://github.com/user-attachments/assets/063c5763-2bd1-475d-831c-af0b4ec1af31)

In addition when I run the simulations and I plot I have straight lines from 1 sec to 0 sec of the simulation. Has anybody had a similar issue?

**Reply by Kostas-Tsak** [reply]
From the queuelength:vector i get this plot that doesn't look exactly as the one in the example
![image](https://github.com/user-attachments/assets/6357dd1f-396b-4a0a-be1a-b31da601f882)
